lib/trino-hive-formats/src/main/java/io/trino/hive/formats/line/json/JsonSerializer.java

```diff
@@ -123,7 +123,16 @@ else if (REAL.equals(type)) {
             return (generator, block, position) -> generator.writeNumber(REAL.getFloat(block, position));
         }
         else if (DOUBLE.equals(type)) {
-            return (generator, block, position) -> generator.writeNumber(DOUBLE.getDouble(block, position));
+            return (generator, block, position) -> {
```

Comment:
What about REAL type?

Reply:
'NaN' and 'Infinity' getting casted as Double it seems. It goes through double datatype.

Reply:
I don't think so. Please try `create table test_nan with (format = 'json') as select real 'NaN' a;`.

Reply:
okay